Make build user support a single build environment

dav3r opened the pull request:
# <!-- Use the title to describe PR changes in the imperative mood --> #

## 🗣 Description ##

This PR modifies the build user that is created by this module such that the user can only build AMIs in a single environment.

<!-- Describe the "what" of your changes in detail. -->
<!-- To avoid scope creep, limit changes to a single goal. -->

## 💭 Motivation and context ##

In the past, we had a single build user that was capable of building AMIs in both a staging and production environment.  In our target (near-future) state, each packer repo will have one build user per environment (e.g. dev, staging, production, etc.)

This supports our overall goal of a COOL system where each environment's IAM users exist in an AWS account specific to that environment and no others.

> [!NOTE] 
> I am marking this PR as "blocked" until the related skeleton-packer PR (https://github.com/cisagov/skeleton-packer/pull/393) has been reviewed and approved, to minimize the amount of time that our cisagov packer repos are out of sync with the latest changes in this module. 

<!-- Why is this change required? -->
<!-- What problem does this change solve? How did you solve it? -->
<!-- Mention any related issue(s) here using appropriate keywords such -->
<!-- as "closes" or "resolves" to auto-close them on merge. -->

## 🧪 Testing ##

I used the version of the module in this PR to successfully create dev, staging, and production build users for https://github.com/cisagov/skeleton-packer/pull/393.

<!-- How did you test your changes? How could someone else test this PR? -->
<!-- Include details of your testing environment, and the tests you ran to -->
<!-- see how your change affects other areas of the code, etc. -->

<!--
## 📷 Screenshots (if appropriate) ##

Uncomment this section if a screenshot is needed.

-->

## ✅ Pre-approval checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- Draft PRs should have one or more unchecked boxes. -->
<!-- If you're unsure about any of these, don't hesitate to ask. -->
<!-- We're here to help! -->

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] All new and existing tests pass.

## ✅ Post-merge checklist ##

<!-- Remove any of the following that do not apply. -->

- [ ] Update all cisagov *-packer repos to reflect the changes in this PR (see https://github.com/cisagov/skeleton-packer/pull/393)
- [x] Create/update dev/staging/prod build users for all cisagov *-packer repos

dav3r committed:
Make providers environment-agnostic

We no longer want providers for Production and Staging because our planned future state prevents Production and Staging accounts from co-mingling in the same AWS organization.

dav3r committed:
Stop using cisagov/ci-iam-user-tf-module and just create the user and AMI-creation role in this module

cisagov/ci-iam-user-tf-module is no longer useful in our target state with separate Production and Staging environments.

dav3r committed:
Additional unwinding of production and staging resources

Also, make some more changes related to the removal of cisagov/ci-iam-user-tf-module.

dav3r committed:
Fix a typo (duplicated word) in a variable description

dav3r committed:
Enable assumption of the AMI creation role

dav3r committed:
Update README to reflect recent changes

dav3r committed:
Update example to reflect recent changes

jsf9k commented on README.md:
```diff
@@ -41,41 +39,40 @@ module "example" {
 |------|---------|
 | aws | >= 4.9 |
-| aws.images-production-ami | >= 4.9 |
-| aws.images-staging-ami | >= 4.9 |
+| aws.images-ami | >= 4.9 |
```
Why is the `aws.images-ssm` provider not listed here?

dav3r replied:
I'm not sure.  If I add it manually, then re-run `terraform-docs`, it gets removed by `terraform-docs`.

jsf9k replied:
Odd

dav3r replied:
I manually added it with https://github.com/cisagov/ami-build-iam-user-tf-module/pull/62/commits/fc7a10f9d97f49480ce6062f01bb36b18390b15f with the hopes that a future version of `terraform-docs` will fix this.

jsf9k reviewed (approved):
Approved, with one question.

dav3r committed:
Add images-ssm provider

For unknown reasons, terraform-docs is not including this provider.  Re-running terraform-docs will remove it, but that change should be ignored.

Co-authored-by: Jeremy Frasier <jeremy.frasier@gwe.cisa.dhs.gov>

mcdonnnj commented on user.tf:
```diff
@@ -2,25 +2,16 @@
 # ------------------------------------------------------------------------------
 
-module "ci_user" {
-  source = "github.com/cisagov/ci-iam-user-tf-module"
```
I would like to see us update this module and revert back to it as a long term plan. I think having a consolidated module for creating CI users still has value even after we finish breaking up environments. As far as this PR I'm simply suggesting an issue + comment to switch back when feasible.

dav3r replied:
Agreed - please take a look at #63 and https://github.com/cisagov/ami-build-iam-user-tf-module/pull/62/commits/880de23420912f33ac9b18ae06a444df8cdf3249.

mcdonnnj reviewed (approved):
LGTM with minor suggestion.

dav3r committed:
Add a TODO comment

TODO: Revert to using cisagov/ci-iam-user-tf-module after it has been updated to create single-environment IAM users.  For details, see: https://github.com/cisagov/ami-build-iam-user-tf-module/issues/63

Co-authored-by: mcdonnnj <mcdonnnj@github.com>
